src/test/java/com/exasol/adapter/request/parser/DataTypeParserTest.java

```diff
@@ -0,0 +1,261 @@
+                        entry("withlocaltimezone", true)), //
+                group(entry("type", "BOOLEAN")), //
+                group(entry("type", "GEOMETRY"), //
+                        entry("scale", 42)), //
+                group(entry("type", "INTERVAL"), //
+                        entry("precision", 32), //
+                        entry("fraction", 51)), //
+                group(entry("type", "HASHTYPE")));
+        final List<DataType> expected = List.of( //
+                DataType.createVarChar(21, ExaCharset.ASCII), //
+                DataType.createChar(22, ExaCharset.UTF8), //
+                DataType.createDecimal(31, 41), //
+                DataType.createDouble(), //
+                DataType.createDate(), //
+                DataType.createTimestamp(true), //
+                DataType.createBool(), //
+                DataType.createGeometry(42), //
+                DataType.createIntervalDaySecond(32, 51), //
+                DataType.createHashtype(16));
+        assertThat(parse(builder), equalTo(expected));
+    }
+
+    private void verifyIllegalPropertyValue(final String datatype, final String property, final String value) {
+        verifyIllegalPropertyValue(group(entry("type", datatype), entry(property, value)), datatype, property,
+                String.valueOf("\"" + value + "\""));
+    }
+
+    private void verifyIllegalPropertyValue(final String datatype, final String property, final boolean value) {
+        verifyIllegalPropertyValue(group(entry("type", datatype), entry(property, value)), datatype, property,
+                String.valueOf(value));
+    }
+
+    private void verifyIllegalPropertyValue(final String datatype, final String property, final int value) {
+        verifyIllegalPropertyValue(group(entry("type", datatype), entry(property, value)), datatype, property,
+                String.valueOf(value));
+    }
+
+    private void verifyIllegalPropertyValue(final JsonParent builder, final String datatype, final String property,
+            final String value) {
+        final Exception e = assertThrows(DataTypeParserException.class, () -> parse(array(builder)));
+        assertThat(e.getMessage(), startsWith("E-VSCOMJAVA-40: Datatype '" + datatype + "', property '" + property
+                + "': Illegal value " + value + "."));
+    }
+
+    // --------------------------------------------
+    // test utilities
+
+    private void verifyMissingRequiredProperty(final JsonParent builder, final String datatype, final String property) {
+        final Exception e = assertThrows(DataTypeParserException.class, () -> parse(array(builder)));
+        assertThat(e.getMessage(),
+                startsWith("E-VSCOMJAVA-36: Datatype '" + datatype + "': Missing property '" + property));
+    }
+
+    private void verifySingle(final DataType expected, final JsonEntry builder) {
+        final DataType actual = parse(array(builder)).get(0);
+        assertThat(actual, equalTo(expected));
```

Comment:
Add an EqualsVerifier test for `DataType` to `DataTypeTest` to ensure that equals works as expected